sw.js

```diff
@@ -0,0 +1,120 @@
+const url = "ws://localhost:8087"
+let socket;
+
+// Open socket connection to SDL
+function connectToSocket() {
+    return new Promise(resolve => {
+        if(!socket || socket.readyState === socket.CLOSED) {
+            socket = new WebSocket(url);
+            socket.onopen = () => {
+                resolve('Connected to SDL socket');
+            }
+            socket.onmessage = onSocketMessage;
+        } else {
+            resolve('Existing socket connection');
+        }
+    });
+}
+
+// Service Worker install event
+function handleInstall(event) {
+    // not caching anything
+    event.waitUntil(connectToSocket());
+}
+
+// Service Worker Activate event
+function handleActivate(event) {
+    // Claim all clients currently in scope
+    event.waitUntil(self.clients.claim());
+}
+
+// Send message to socket connection
+function sendToSocket(rpc) {
+    socket.send(rpc);
+}
+
+function disconnectFromSocket() {
+    if (socket) {
+        if(socket.readyState === socket.OPEN) {
+            socket.onclose = function () {
+                socket.close();
```

Comment:
Does this actually close the socket?  Looks more like it sets up an onclose handler.